geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/data/KafkaDataStoreTest.scala

```diff
@@ -188,6 +192,86 @@ class KafkaDataStoreTest extends KafkaContainerTest with Mockito {
       }
     }
 
+    "allow schemas to be created and truncated" >> {
+
+      foreach(Seq(true, false)) { zk =>
+        TableBasedMetadata.Expiry.threadLocalValue.set("10ms")
+        val (producer, consumer, _) = try {
+          createStorePair(if (zk) {
+            Map(
+              "kafka.zookeepers" -> zookeepers,
+              "kafka.catalog.on-schema-delete-truncate" -> "true"
+            )
+          } else {
+            Map(
+              "kafka.catalog.on-schema-delete-truncate" -> "true"
+            )
+          })
+        } finally {
+          TableBasedMetadata.Expiry.threadLocalValue.remove()
+        }
+        consumer must not(beNull)
+        producer must not(beNull)
+
+        try {
+          val sft = SimpleFeatureTypes.createImmutableType("kafka", "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;geomesa.foo='bar'")
+          val topic = s"${producer.config.catalog}-${sft.getTypeName}".replaceAll("/", "-")
+
+          producer.createSchema(sft)
+
+          consumer.metadata.resetCache()
+          foreach(Seq(producer, consumer)) { ds =>
+            ds.getTypeNames.toSeq mustEqual Seq(sft.getTypeName)
+            val schema = ds.getSchema(sft.getTypeName)
+            schema must not(beNull)
+            schema mustEqual sft
+            schema.getUserData.get("geomesa.foo") mustEqual "bar"
+            schema.getUserData.get(KafkaDataStore.TopicKey) mustEqual topic
+          }
```

Comment:
just so that we're not testing the format of the data topic, which is an internal algorithm that shouldn't fail this test if it changes:

```suggestion
          producer.createSchema(sft)
          consumer.metadata.resetCache()

          foreach(Seq(producer, consumer)) { ds =>
            ds.getTypeNames.toSeq mustEqual Seq(sft.getTypeName)
            val schema = ds.getSchema(sft.getTypeName)
            schema must not(beNull)
            schema mustEqual sft
          }
          val topic = producer.getSchema(sft.getTypeName).getUserData.get(KafkaDataStore.TopicKey)
```